enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,340 @@
+
+```console
+$ oc get machineconfignodes -o wide
+NAME                          UPDATED   UPDATEPREPARING   UPDATEINPROGRESS   UPDATEPOSTACTION   UPDATECOMPLETING   RESUMING   COMPARINGMACHINECONFIGS   UPDATINGFILESANDOS   DRAININGNODE   CORDONINGNODE   REBOOTINGNODE   RELOADINGCRIO
+ip-10-0-1-37.ec2.internal     True      False             False              False              False              False      False                     False                False          False           False           False
+ip-10-0-33-243.ec2.internal   True      False             False              False              False              False      False                     False                False          False           False           False
+ip-10-0-37-208.ec2.internal   True      False             False              False              False              False      False                     False                False          False           False           False
+ip-10-0-39-225.ec2.internal   False     False             False              True               False              False      False                     False                False          False           True            False
+ip-10-0-51-162.ec2.internal   True      False             False              False              False              False      False                     False                False          False           False           False
+ip-10-0-59-150.ec2.internal   False     False             False              True               False              False      False                     False                False          False           True            False
+```
+
+Where each name represents a node in a given machineconfigpool. The statuses reported are created explicitly from MCO node annotations and MCO actions, no other operator actions are taken into account here. This allows us to get quite specific in what is occuring on the nodes.
+
+```console
+Name:         ip-10-0-12-194.ec2.internal
+Namespace:    
+Labels:       <none>
+Annotations:  <none>
+API Version:  machineconfiguration.openshift.io/v1alpha1
+Kind:         MachineConfigNode
+Metadata:
+  Creation Timestamp:  2023-10-17T13:08:58Z
+  Generation:          1
+  Resource Version:    49443
+  UID:                 4bd758ab-2187-413c-ac42-882e61761b1d
+Spec:
+  Node Ref:
+    API Version:  
+    Kind:         Node
+    Name:         ip-10-0-12-194.ec2.internal
+  Pool:
+    API Version:  machineconfiguration.openshift.io/v1
+    Kind:         MachineConfigPool
+    Name:         master
+Status:
+  Conditions:
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               Node has completed update to config rendered-master-cf99e619747ab19165f11e3546c71f1e
+    Reason:                NodeUpgradeComplete
+    Status:                True
+    Type:                  Updated
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               This node has not yet entered the UpdatePreparing phase
```

Comment:
The message `This node has not yet entered the UpdatePreparing phase` contradicts with `Reason: NodeUpgradeComplete`

Reply:
this is not contradictory, the status is false. the update has not yet entered this phase. All phases must be accounted for or else the CR cannot be updated.